pkg/bindings/http/sender.go

```diff
@@ -45,10 +53,40 @@ func (s *Sender) Send(ctx context.Context, m binding.Message) (err error) {
 	return
 }
 
-func NewSender(client *http.Client, target *url.URL, options ...SenderOptionFunc) binding.Sender {
-	s := &Sender{Client: client, Target: target, transformerFactories: make(binding.TransformerFactories, 0)}
-	for _, o := range options {
-		o(s)
+func (r *Sender) Request(ctx context.Context, m binding.Message) (binding.Receiver, error) {
+	var err error
+	defer func() { _ = m.Finish(err) }()
+	if r.Client == nil || r.Target == nil {
+		return nil, fmt.Errorf("not initialized: %#v", r)
 	}
-	return s
+
+	var req *http.Request
+	req, err = http.NewRequest("POST", r.Target.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req = req.WithContext(ctx)
+
+	if err = EncodeHttpRequest(ctx, m, req, r.transformerFactories); err != nil {
+		return nil, err
+	}
+	resp, err := r.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("%d %s", resp.StatusCode, nethttp.StatusText(resp.StatusCode))
+	}
+
+	msg, err := NewMessage(resp.Header, resp.Body)
+	return &cachedReceiver{Message: msg, Error: err}, err
+}
+
+type cachedReceiver struct {
+	Message binding.Message
+	Error   error
+}
+
+func (r *cachedReceiver) Receive(ctx context.Context) (binding.Message, error) {
```

Comment:
Now that I see it I frankly don't like it. The contract of `Receiver` gives the idea that every time you invoke `Receive`, or you receive an `io.EOF`, or an error, or a new `Message`, but that's not what happens here...

Maybe should we rework the `Requester` interface to just let `Request` return a `Message`?

Reply:
The only downside is that if you have return Message, what do you do when there is a batch or stream of events?

Reply:
Do we plan somehow to support batch? Because I have no clue on how to bring it inside binding APIs now :fearful: